Signal/src/ViewControllers/ProfileViewController.m

```diff
@@ -223,22 +223,40 @@ - (void)updatePressed
 - (void)updateProfile
 {
     __weak ProfileViewController *weakSelf = self;
-    [OWSProfileManager.sharedManager updateLocalProfileName:self.nameTextField.text
-        avatarImage:self.avatar
-        success:^{
-            [weakSelf.navigationController popViewControllerAnimated:YES];
-        }
-        failure:^{
-            [OWSAlerts showAlertWithTitle:NSLocalizedString(@"ALERT_ERROR_TITLE", @"")
-                                  message:NSLocalizedString(@"PROFILE_VIEW_ERROR_UPDATE_FAILED",
-                                              @"Error message shown when a profile update fails.")];
-        }];
+
+    // Show an activity indicator to block the UI during the profile upload.
+    UIAlertController *alertController =
+        [UIAlertController alertControllerWithTitle:NSLocalizedString(@"PROFILE_VIEW_SAVING",
+                                                        @"Indicates that the user's profile view is being saved.")
+                                            message:nil
+                                     preferredStyle:UIAlertControllerStyleAlert];
+
+    [self presentViewController:alertController
+                       animated:YES
+                     completion:^{
+                         [alertController dismissViewControllerAnimated:NO completion:nil];
```

Comment:
Am I understanding that we present the "saving" alert only momentarily, then immediately dismiss it. Then start uploading.

I would have thought the `OWSProfileManager.sharedManager updateLocalProfileName` completion handler would dismiss the alert, otherwise it will just be a quick flash right?